src/app/lightning/update-ln-payments.ts

```diff
@@ -0,0 +1,90 @@
+import { LndService } from "@services/lnd"
+import { LnPaymentsRepository } from "@services/mongoose/ln-payments"
+
+export const updateLnPayments = async (): Promise<true | ApplicationError> => {
+  let incompleteLnPayments = await LnPaymentsRepository().listIncomplete()
+  if (incompleteLnPayments instanceof Error) return incompleteLnPayments
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+
+  const lndListMethods = [lndService.listSettledPayments, lndService.listFailedPayments]
+  for (const listPaymentsFn of lndListMethods) {
+    incompleteLnPayments = await fetchAndUpdatePayments({
+      incompleteLnPayments,
+      listPaymentsFn,
+    })
+    if (incompleteLnPayments instanceof Error) return incompleteLnPayments
+    if (incompleteLnPayments.length == 0) return true
+  }
+
+  return true
+}
+
+const fetchAndUpdatePayments = async ({
+  incompleteLnPayments,
+  listPaymentsFn,
+}: {
+  incompleteLnPayments: PersistedLnPaymentLookup[]
+  listPaymentsFn: ({
+    after,
+    pubkey,
+  }: ListLnPaymentsArgs) => Promise<ListLnPaymentsResult | LightningServiceError>
+}) => {
+  const processedLnPaymentsHashes: PaymentHash[] | ApplicationError = []
+  const incompleteLnPaymentsHashes = incompleteLnPayments.map((p) => p.paymentHash)
+
+  const lndService = LndService()
+  if (lndService instanceof Error) return lndService
+  const pubkeys = lndService.listPubkeys()
```

Comment:
Should we be listing only active lnds here, or is it ok to include all lnds?

Reply:
if an lnd is offline we won't be able to do much with it